### ISSUE TITLE
tsc 1.5.3 can't find a module, even though Visual Studio Code has no problems with it.

### ISSUE THREAD

**Issue by jedmao** [Microsoft/TypeScript#4134]
Running the following command with Git Bash on Windows:

```
15:56 $tsc --version
message TS6029: Version 1.5.3
15:56 $tsc
typings/postcss/.d.ts(2,22): error TS2307: Cannot find module 'd.ts/postcss'.
```

`tsc` can't find a module, even though it's specified in `tsconfig.json` under `files` like so:

```json
{
  "compilerOptions": {
    "target": "es6",
    "declaration": true,
    "noResolve": true,
    "outDir": "build",
    "rootDir": "."
  },
  "files": [
    "typings/postcss/.d.ts",
    "lib/plugin.ts"
  ]
}
```
Visual Studio Code doesn't seem to mind:

![visual-studio-code](https://cloud.githubusercontent.com/assets/1058243/9047681/17f55948-39f9-11e5-960e-c559e47e4554.png)

Also, it gives me Intellisense just fine:

![intellisense](https://cloud.githubusercontent.com/assets/1058243/9047709/435b8efe-39f9-11e5-92f8-60c4c6fbe235.png)

So why does `tsc` have a problem with it?

**Comment by mhegazy** [comment]
i could not get a repro for this on latest bits. can you give `typescript@next` a try, or post more details. i suspect there is more going on than i could reconstruct from the sample above. the file name `.d.ts` seems to be something that we would easily get wrong. though i could not prove it :). also any investigation in the root cause would be appreciated.

**Comment by jedmao** [comment]
Turns out I needed to tell `tsconfig.json` not just `typings/postcss/.d.ts`, but also the 20 or so files that that file required as well. >_<

Thanks.

**Comment by jedmao** [comment]
@mhegazy I have more information concerning this issue.

I turned `noResolve` to `false` in my [`tsconfig.json`][] by removing the option entirely, because I don't want to reference 20+ definition files manually in my `files` array:

```json
{
  "compilerOptions": {
    "target": "es6",
    "declaration": true,
    "newLine": "LF",
    "outDir": "build",
    "rootDir": "."
  },
  "files": [
    "lib/plugin.ts",
    "test/plugin.ts"
  ]
}
```

Running `tsc` with the above [`tsconfig.json`][] gives me the following error:

> typings/postcss/.d.ts(2,22): error TS2307: Cannot find module 'd.ts/postcss'.

Strangely, adding only `"typings/postcss/d.ts/postcss.d.ts"` and no other definition files to my `files` array [fixes the issue entirely](https://github.com/jedmao/postcss-triangle/blob/6b428fd0f89011373858db02813bff6f554293a8/tsconfig.json), but I shouldn't have to do this with `noResolve: false`, right?

Is there something inherently wrong w/ [the way I'm referencing definitions here?](https://github.com/jedmao/postcss-triangle/blob/6b428fd0f89011373858db02813bff6f554293a8/typings/postcss/.d.ts)

![visual-studio-code](https://cloud.githubusercontent.com/assets/1058243/9047681/17f55948-39f9-11e5-960e-c559e47e4554.png)

It's an open source project, so [you can clone](https://github.com/jedmao/postcss-triangle) and play with it however you want to reproduce.

[`tsconfig.json`]: https://github.com/jedmao/postcss-triangle/blob/6b428fd0f89011373858db02813bff6f554293a8/tsconfig.json

**Comment by mhegazy** [comment]
I have not looked into this since i last posted. my apologies. is this still an issue?

**Comment by jedmao** [comment]
It looks like it's fixed in the newest version, `Version 1.8.0-dev.20151113`. Thanks!

**Comment by mhegazy** [comment]
thanks @jedmao for confirming.

**Comment by kdcllc** [comment]
@mhegazy 
Here is the commit from the project that I experienced the same kind of issue: https://github.com/kdcllc/Angular-TypeScript-Starter/commit/f5e0ff23e634ed07b1c2152d79bff2f29b312b64
It seems that I was able to fix it by adding custom.d.ts reference file. I believe that typescript engine reads thru directories to load a proper typing for the builds. I have other projects that don't have the d.ts reference for the custom definition files and it works.
